src/components/Regridder.vue

```diff
@@ -0,0 +1,232 @@
+import MapboxDraw from '@mapbox/mapbox-gl-draw'
+import DrawRectangle from 'mapbox-gl-draw-rectangle-mode'
+import {DownloadControl} from "../lib/DownloadControl"
+import { PiWebserviceProvider} from "@deltares/fews-pi-requests";
+import PiRequestsMixin from '@/mixins/PiRequestsMixin';
+
+@Component
+export default class Regridder extends Mixins(PiRequestsMixin) {
+    @Inject() getMap!: () => Map
+
+    @Prop() firstValueTime!: string
+    @Prop() lastValueTime!: string
+
+    mapObject!: Map
+    isInitialized = false
+    draw!: MapboxDraw
+    downloadControl: DownloadControl | undefined
+    downloadDialog = false
+    errorDialog = false
+    dx = ''
+    dy = ''
+    errorMessage = 'Select an area on the map before downloading the data.'
+    bbox: number[] | undefined
+    webServiceProvider!: PiWebserviceProvider
+    baseUrl: string | undefined
+
+    created(): void {
+        this.baseUrl = this.$config.get('VUE_APP_FEWS_WEBSERVICES_URL')
+        const transformRequestFn = this.getTransformRequest()
+        this.webServiceProvider = new PiWebserviceProvider(this.baseUrl, {transformRequestFn})
+    }
+        
+    deferredMountedTo(map: Map) {
+        this.mapObject = map
+        this.mapObject.dragRotate.disable()
+        this.mapObject.touchZoomRotate.disableRotation()
+        this.mapObject.once('load', () => {
+        this.isInitialized = true
+        this.addToMap()
+    })
+    }
+
+    refreshDownloadControl(bbox: number[] | undefined ) {
+        if (this.downloadControl){
+            this.mapObject.removeControl(this.downloadControl)
+        }
+        this.downloadControl = new DownloadControl(bbox, this)
+        this.mapObject.addControl(this.downloadControl)
+    }
+
+    select() {
+        const data = this.draw.getAll()
+        // get bbox from data
+        const geometry: any = data.features[0].geometry // change any  to proper type
+        if (geometry.coordinates !== null) {
+            const coordinates = geometry.coordinates[0]
+            const x = coordinates.map((c: any) => c[0])
+            const y = coordinates.map((c: any) => c[1])
+            const xmin = Math.min(...x)
+            const xmax = Math.max(...x)
+            const ymin = Math.min(...y)
+            const ymax = Math.max(...y)
+            this.bbox = [xmin, ymin, xmax, ymax]
+            this.refreshDownloadControl(this.bbox)
+        }
+    }
+
+    addToMap() {
+        this.draw = new MapboxDraw({
+        displayControlsDefault: false,
+        controls: {
+            polygon: true,
+            trash: true,
+        },
+        modes:  Object.assign(MapboxDraw.modes, {
+            draw_rectangle: DrawRectangle,
+        } as any),
+        defaultMode: 'draw_rectangle',
+        })
+
+        this.mapObject.addControl(this.draw)
+
+        this.draw.changeMode("draw_rectangle")
+
+        this.mapObject.on('draw.create', this.select)
+
+        const trashElement = document.querySelector(".mapbox-gl-draw_trash")
```

Comment:
Select on `id` rather than `class`? `id` should always be unique, while multiple elements may share the same `class`

Reply:
unfortunately that doesn't seem to be possible since the trashElement is defined by mapbox